src/main/java/org/opensearch/knn/index/vectorvalues/KNNBinaryVectorValues.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.vectorvalues;
+
+import lombok.ToString;
+import org.apache.lucene.codecs.KnnFieldVectorsWriter;
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.ByteVectorValues;
+
+/**
+ * Concrete implementation of {@link KNNVectorValues} that returns byte[] as vector where binary vector is stored and
+ * provides an abstraction over {@link BinaryDocValues}, {@link ByteVectorValues}, {@link KnnFieldVectorsWriter} etc.
+ */
+@ToString(callSuper = true)
+public class KNNBinaryVectorValues extends KNNByteVectorValues {
```

Comment:
Personally, it looks better to understand the code by making KNNBinaryVectorValues in the same hierarchical level with other Byte and Float even if we have to repeat the getVector() method. However, I will just leave it to you to make a final decision. Avoiding a repetition of code is not a thing we should pursue the most imo.

```suggestion
public class KNNBinaryVectorValues extends KNNVectorValues<byte[]> {
```

Reply:
Let me try to see if I can remove code duplication by keeping them at same level. I can see BitVectorFormat at same level as the byteVectors format. I will change it.

https://github.com/apache/lucene/tree/main/lucene/codecs/src/java/org/apache/lucene/codecs/bitvectors